src/pres/loader/PGNLoader.js

```diff
@@ -208,6 +211,43 @@ export default class PGNLoader extends React.Component {
         this.props.variantChange(newVariant)
         trackEvent(Constants.EVENT_CATEGORY_PGN_LOADER, "VariantChange", newVariant)
     }
+    getInitiallyExpandedPanel(selectedSite, playerName) {
+        if (!selectedSite) {
+            return 'source'
+        } else if (!playerName) {
+            return 'user'
+        } else {
+            return 'filters'
+        }
+    }
+    getTimeControls() {
+        let selectedChoices = new URLSearchParams(window.location.search).get("timeControls")
+
+        const allChoices = [
+            Constants.TIME_CONTROL_ULTRA_BULLET,
+            Constants.TIME_CONTROL_BULLET,
+            Constants.TIME_CONTROL_BLITZ,
+            Constants.TIME_CONTROL_RAPID,
+            Constants.TIME_CONTROL_CLASSICAL,
+            Constants.TIME_CONTROL_CORRESPONDENCE,
+            Constants.TIME_CONTROL_DAILY
+        ]
+
+        if (!selectedChoices) {
+            const result = {}
+            for (const choice of allChoices) {
+                result[choice] = true
+            }
+            return result
+        } else {
+            selectedChoices = selectedChoices.split(',')
+            const result = {}
+            for (const choice of allChoices) {
+                result[choice] = selectedChoices.includes(choice)
+            }
+            return result
```

Comment:
`getTimeControls` parses a comma-separated list of time controls. Probably could use a  query params library for this instead, if you want.